tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,365 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+import os
+
+import mmcv
+import pytest
+from mmcv.utils import Registry
+from torch import nn
+
+from otx.cli.builder.builder import (
+    Builder,
+    get_backbone_out_channels,
+    update_backbone_args,
+    update_channels,
+)
+from otx.cli.utils.importing import get_otx_root_path
+from otx.mpa.utils.config_utils import MPAConfig
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class TestOTXCLIBuilder:
+    @e2e_pytest_unit
+    def test_builder_build_task_config(self, tmp_dir_path):
+        """
+        <b>Description:</b>
+        Check "Builder.build_task_config" function that create otx-workspace is working well
+
+        <b>Steps</b>
+        1. Create Classification custom workspace
+        2. Raising Error of building workspace with already created path
+        3. Update hparam.yaml with train_type="selfsl"
+        4. Raising ValueError with wrong train_type
+        5. Build workspace with model_type argments
+        6. Raise ValueError when build workspace with wrong model_type argments
+        """
+        otx_builder = Builder()
```

Comment:
```
otx_builder = Builder()
otx_root = get_otx_root_path()
```
how about making these parts as fixtures?

Reply:
There is a question as to whether that is correct. Those objects are not considered general objects that are only used in a small number of test cases.
I'm afraid if I subtract this as fixtures, end up with too many fixtures later. I think we need to narrow the criteria down to something more general.

Reply:
This is also bundled as an internal fixture of the class. https://github.com/openvinotoolkit/training_extensions/pull/1519/commits/fc590a79fddc0d6c695a186f904762ae4aea5f92